runtime/lua/vim/filetype.lua

```diff
@@ -2049,6 +2051,7 @@ local pattern = {
   ['XF86Config%-4.*'] = starsetf(detect.xfree86_v4, { priority = -math.huge + 1 }),
   ['XF86Config.*'] = starsetf(detect.xfree86_v3),
   ['%.zcompdump.*'] = starsetf('zsh'),
+  ['%.zsh_history'] = starsetf('zsh'),
```

Comment:
This should be in the `filename` table and should not use `starsetf()`.

Reply:
Why this one and not the one above?

Reply:
(`starsetf`, that is)

Reply:
1. It's not used in Vim.
2.
```vim
" Function used for patterns that end in a star: don't set the filetype if the
" file name matches ft_ignore_pat.
" When using this, the entry should probably be further down below with the
" other StarSetf() calls.
func s:StarSetf(ft)
  if expand("<amatch>") !~ g:ft_ignore_pat
    exe 'setf ' . a:ft
  endif
endfunc
```